pkg/types/server_types.go

```diff
@@ -229,16 +236,24 @@ type APIEvent struct {
 	Data interface{} `json:"data,omitempty"`
 }
 
+type Warning struct {
+	Code  int
+	Agent string
```

Comment:
I don't see this being used here or in the steve PR, is it necessary?

Reply:
Yeah, the steve side does keep a slice of these in the "WarningBuffer"

Reply:
I see it adding it to the slice https://github.com/rancher/steve/pull/69/files#diff-82b79befb09b4c7baa9a34e0412389dc2be544a3e56086726419bbf939a9e9b9R74 but where does it actually use it? Only the Code and Text are used in the warning message string.

Reply:
Upon further review, the "agent" is part of the spec for warning headers, even though it's likely to be set to "-", I think we should keep it.  I will add it to what is logged just in case it ever proves to be useful.